src/core/memory.cc

```diff
@@ -1121,20 +1121,19 @@ cpu_pages::try_free_fastpath(void* ptr) {
 }
 
 
-struct no_size {};
-
-/// Helper to allow a single implementation for sized and non-sized functions.
+/// Indicator to allow a single implementation for sized and non-sized functions.
 /// The size parameter will be either no_size tag type or size_t, and most
 /// of the implementation can be shared, using constexpr if or other dispatch
 /// in the places where there should be a difference of behavior.
-template<typename T>
-concept SizeOrNot = std::same_as<T, size_t> || std::same_as<T, no_size>;
+struct no_size {};
 
+template <typename S>
+SEASTAR_CONCEPT(requires std::same_as<S, size_t> || std::same_as<S, no_size>)
 [[gnu::noinline]]
```

Comment:
Please fold into the patch that introduces the problem.

Reply:
This is addressed in https://github.com/scylladb/seastar/commit/9d5d2445547b0eb47feef73a708dc9aeb1fbe005.